copper-server-mongo/src/main/java/com/pogeyan/cmis/data/mongo/services/MongoClientFactory.java

```diff
@@ -84,8 +89,22 @@
 	private static String MRELATIONOBJECCTDAOIMPL = "MRelationObjectDAOImpl";
 	private Map<Class<?>, String> objectServiceClass = new HashMap<>();
 	private final Map<String, Datastore> clientDatastores = new HashMap<String, Datastore>();
-	private final Map<String, MongoClient> mongoClient = new HashMap<String, MongoClient>();
 	private Morphia morphia = new Morphia();
+	private final static Cache<String, MongoClient> mongoClient;
+	private static RemovalListener<String, MongoClient> removalListener = new RemovalListener<String, MongoClient>() {
+		public void onRemoval(RemovalNotification<String, MongoClient> removal) {
+			MongoClient conn = removal.getValue();
+			LOG.info("Closing MongoClient connection in RemovalListener");
+			conn.close();

```

Comment:
Why are we not removing the clientDatastore from this?

Reply:
we are removing in close function